src/blight/actions/bitcode_extract.py

```diff
@@ -0,0 +1,40 @@
+"""
+The `BitcodeExtract` action.
+"""
+
+import logging
+import os
+import subprocess
+from typing import List
+
+from blight.action import CompilerAction
+from blight.enums import Lang
+from blight.tool import CompilerTool
+
+logger = logging.getLogger(__name__)
+
+
+class BitcodeExtract(CompilerAction):
+    """
+    Compile and extract bitcode. The output bitcode file will be located in the same direcory
+    as the output executable.
+    """
+
+    def before_run(self, tool: CompilerTool) -> None:  # type: ignore
+        if tool.lang in [Lang.C, Lang.Cxx]:
+            args: List[str]
+            if tool.outputs:
+                args = ["-c", "-emit-llvm", "-o", tool.outputs[0] + ".bc", *tool.inputs]
+            else:
+                logger.debug(
+                    "not extracting bitcode with unspecified output location"
+                )  # pragma: no cover
+                assert False  # pragma: no cover
```

Comment:
This should probably just be a `return`, since we want to silently skip this action instead of erroring when it doesn't have anything meaningful to perform.